File: VB/DXApplication7/DesignerForm1.vb
Imports DevExpress.DashboardCommon
Imports System
Imports System.Xml.Linq

Namespace WinDesignerDashboardState
	Partial Public Class DesignerForm1
		Inherits DevExpress.XtraBars.Ribbon.RibbonForm

		Private dState As New DashboardState()
		Private Const path As String = "..\..\Dashboards\dashboardWithSavedState.xml"
		Public Sub New()
			InitializeComponent()
			dashboardDesigner.CreateRibbon()
			dashboardDesigner.LoadDashboard(path)
		End Sub
		Private Sub dashboardDesigner_DashboardLoaded(ByVal sender As Object, ByVal e As DevExpress.DashboardWin.DashboardLoadedEventArgs) Handles dashboardDesigner.DashboardLoaded
			Dim data As XElement = e.Dashboard.UserData
			If data IsNot Nothing Then
				If data.Element("DashboardState") IsNot Nothing Then
					Dim dStateDocument As XDocument = XDocument.Parse(data.Element("DashboardState").Value)
					dState.LoadFromXml(XDocument.Parse(data.Element("DashboardState").Value))
				End If
			End If
		End Sub
		Private Sub dashboardDesigner_SetInitialDashboardState(ByVal sender As Object, ByVal e As DevExpress.DashboardWin.SetInitialDashboardStateEventArgs) Handles dashboardDesigner.SetInitialDashboardState
			e.InitialState = dState
		End Sub

		Private Sub dashboardDesigner_DashboardClosing(ByVal sender As Object, ByVal e As DevExpress.DashboardWin.DashboardClosingEventArgs) Handles dashboardDesigner.DashboardClosing
			dState = dashboardDesigner.GetDashboardState()
			Dim userData As New XElement("Root", New XElement("DateModified",DateTime.Now), New XElement("DashboardState",dState.SaveToXml().ToString(SaveOptions.DisableFormatting)))
			dashboardDesigner.Dashboard.UserData = userData
			dashboardDesigner.Dashboard.SaveToXml(path)
		End Sub
	End Class
End Namespace

File: CS/DXApplication7/DesignerForm1.cs

using System;
using System.ComponentModel;
using System.Xml.Linq;
using DevExpress.DashboardCommon;

namespace WinDesignerDashboardState
{
    public partial class DesignerForm1: DevExpress.XtraBars.Ribbon.RibbonForm {

        DashboardState dState = new DashboardState();
        const string path = @"..\..\Dashboards\dashboard1.xml";
        public DesignerForm1() {
            InitializeComponent();
            dashboardDesigner.CreateRibbon();
            dashboardDesigner.LoadDashboard(path);
        }

        protected override void OnClosing(CancelEventArgs e) {
            base.OnClosing(e);
            dState = dashboardDesigner.GetDashboardState();
            XElement userData = new XElement("Root",
                new XElement("DateModified",DateTime.Now),
                new XElement("DashboardState",dState.SaveToXml().ToString(SaveOptions.DisableFormatting)));
            dashboardDesigner.Dashboard.UserData = userData;
            dashboardDesigner.Dashboard.SaveToXml(path);            
        }
        private void dashboardDesigner_DashboardLoaded(object sender,
            DevExpress.DashboardWin.DashboardLoadedEventArgs e) {
            XElement data = e.Dashboard.UserData;
            if(data != null) {
                if(data.Element("DashboardState") != null) {
                    XDocument dStateDocument = XDocument.Parse(data.Element("DashboardState").Value);
                    dState.LoadFromXml(XDocument.Parse(data.Element("DashboardState").Value));
                }
            }            
        }
        private void dashboardDesigner_SetInitialDashboardState(object sender,
            DevExpress.DashboardWin.SetInitialDashboardStateEventArgs e) {
            e.InitialState = dState;
        }
    }
}
